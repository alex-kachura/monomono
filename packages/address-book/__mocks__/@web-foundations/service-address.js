const original = jest.requireActual('@web-foundations/service-address');

export const mockOn = jest.fn();

const MockAddressService = jest.fn(function AddressServiceMock() {
  Object.assign(this, {
    ...original.default,
    on: mockOn,
  });
});

export default MockAddressService;
