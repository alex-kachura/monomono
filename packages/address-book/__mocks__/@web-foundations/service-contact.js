const original = jest.requireActual('@web-foundations/service-contact');

export const mockOn = jest.fn();
export const mockGetFullAddresses = jest.fn();
export const mockRemoveAddress = jest.fn();

const MockContactService = jest.fn(function ContactServiceMock() {
  Object.assign(this, {
    ...original.default,
    on: mockOn,
    getFullAddresses: mockGetFullAddresses,
    removeAddress: mockRemoveAddress,
  });
});

export const ContactServiceError = original.ContactServiceError;

export default MockContactService;
