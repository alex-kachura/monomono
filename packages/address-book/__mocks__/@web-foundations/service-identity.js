const original = jest.requireActual('@web-foundations/service-identity');

export const mockOn = jest.fn();
export const mockGetClaims = jest.fn().mockReturnValue({ access_token: 'test-access-token' }); // eslint-disable-line camelcase

const MockIdentityService = jest.fn(function IdentityServiceMock() {
  Object.assign(this, {
    ...original.default,
    on: mockOn,
    getClaims: mockGetClaims,
  });
});

MockIdentityService.Environments = original.default.Environments;
MockIdentityService.Endpoints = original.default.Endpoints;
MockIdentityService.StateOfOldCard = original.default.StateOfOldCard;
MockIdentityService.Capabilities = original.default.Capabilities;
MockIdentityService.OAuthTokenGrantTypes = original.default.OAuthTokenGrantTypes;
MockIdentityService.OAuthTokenScopes = original.default.OAuthTokenScopes;
MockIdentityService.Schemas = original.default.Schemas;

export default MockIdentityService;
