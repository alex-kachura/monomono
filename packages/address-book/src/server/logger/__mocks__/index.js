export const mockDebug = jest.fn();
export const mockInfo = jest.fn();
export const mockValidation = jest.fn();
export const mockWarn = jest.fn();
export const mockError = jest.fn();
export const mockMakeOnRequestEventHandler = jest.fn(() => Function);

const mockNodeLogger = jest.fn(() => ({
  debug: mockDebug,
  info: mockInfo,
  validation: mockValidation,
  warn: mockWarn,
  error: mockError,
}));

mockNodeLogger.makeOnRequestEventHandler = mockMakeOnRequestEventHandler;

export default mockNodeLogger;

export const logOutcome = jest.fn((name, outcome, { sessionId, region, lang }) =>
  mockInfo(`address-book:${name}`, {
    outcome,
    tracer: sessionId,
    region,
    lang,
  }),
);
