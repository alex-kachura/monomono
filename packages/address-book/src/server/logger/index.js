import config from 'config';
import winston from 'winston';
import loggerFactory from '@web-foundations/logger';
import { clearTextWhitelist } from '@oneaccount/node-logger-whitelists/lib/whitelists/clear-text';
import { redactedWhitelist } from '@oneaccount/node-logger-whitelists/lib/whitelists/redacted';
import { regexRedactedWhitelist } from '@oneaccount/node-logger-whitelists/lib/whitelists/regex-redacted';
import { messageFilterRegexes } from '@oneaccount/node-logger-whitelists/lib/filters/message';
import preParseRewriterFactory from '@oneaccount/node-logger-whitelists/lib/rewriters/pre-parser';
import clearTextRewriterFactory from '@web-foundations/logger/lib/rewriters/clear-text';
import redactedRewriterFactory from '@web-foundations/logger/lib/rewriters/redacted';
import regexRedactedRewriterFactory from '@web-foundations/logger/lib/rewriters/regex-redacted';
import messageFilterFactory from '@web-foundations/logger/lib/filters/message';

const transports = [
  new winston.transports.Console({
    colorize: process.env.NODE_ENV === 'development',
    datePattern: config.log.datePattern,
    json: true,
    level: process.env.NODE_ENV === 'development' ? 'debug' : 'info',
    prettyPrint: process.env.NODE_ENV === 'development',
    stringify:
      process.env.NODE_ENV === 'development'
        ? (_) => JSON.stringify(_, null, 2)
        : (_) => JSON.stringify(_),
    timestamp: () => new Date().toJSON(),
  }),
];

const nodeEnv = process.env.NODE_ENV;
const logConf = config.log;

const encryptionConfig = {
  key: logConf.encryptionKey,
  disabled: logConf.encryptionDisabled,
};

const clearTextRewriter = clearTextRewriterFactory({
  whitelist: clearTextWhitelist,
});

const redactedRewriter = redactedRewriterFactory({
  whitelist: redactedWhitelist,
  encryptionConfig,
  nodeEnv,
});

const regexRedactedRewriter = regexRedactedRewriterFactory({
  whitelist: regexRedactedWhitelist,
  encryptionConfig,
  nodeEnv,
});

const messageFilter = messageFilterFactory({
  encryptionConfig,
  messageFilterRegexes,
  nodeEnv,
});

const preParseRewriter = preParseRewriterFactory({
  application: config.name,
});

const rewriters = [preParseRewriter, clearTextRewriter, redactedRewriter, regexRedactedRewriter];
const filters = [messageFilter];
const logger = loggerFactory({ transports, rewriters, filters });

logger.makeOnRequestEventHandler = function makeOnRequestEventHandler(message) {
  return function onRequestEvent(data) {
    const { context, ...meta } = data;

    if (meta.startTime && meta.endTime) {
      meta.ms = meta.endTime - meta.startTime;
    }

    logger.info(message, meta, context);
  };
};

export const logOutcome = (name, outcome, { sessionId, region, lang }) =>
  logger.info(`address-book:${name}`, {
    outcome,
    tracer: sessionId,
    region,
    lang,
  });

export default logger;
