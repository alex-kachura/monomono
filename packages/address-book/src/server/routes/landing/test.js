import { mockGetFullAddresses } from '@web-foundations/service-contact';
import { logOutcome } from '../../logger';
import { getLandingPage } from '.';

describe('[Route: /]', () => {
  let responseType;
  const mockTracer = 'mock-tracer';
  const mockUpdated = 'mock-updated';
  const mockLocale = 'mock-locale';
  const mockAccessToken = 'mock-token';
  const mockAddresses = [
    {
      label: 'HOME',
      tags: ['primaryDelivery'],
      addressLines: [
        {
          lineNumber: 1,
          label: 'Lever building',
        },
      ],
      postcode: 'EC1R 5AR',
      postTown: 'London',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '07429049112',
        },
      ],
    },
    {
      label: 'HOME',
      tags: ['primaryLoyalty'],
      addressLines: [
        {
          lineNumber: 1,
          label: 'Lever building',
        },
      ],
      postcode: 'EC1R 5AR',
      postTown: 'London',
      telephoneNumbers: [
        {
          label: 'Day',
          value: '07429049112',
        },
      ],
    },
  ];
  const req = {
    sessionId: mockTracer,
    locale: { type: mockLocale },
    query: { updated: mockUpdated },
    region: 'mock-region',
    lang: 'mock-lang',
    getClaims: jest.fn(() => ({ accessToken: mockAccessToken })),
  };
  const res = {
    format: (formats) => formats[responseType](),
    send: jest.fn(),
    data: {},
  };
  const next = jest.fn();

  beforeAll(() => {
    expect(mockGetFullAddresses.mock).toBeTruthy();
    expect(logOutcome.mock).toBeTruthy();
  });

  describe('[GET]', () => {
    describe('[Content-Type: html]', () => {
      beforeAll(() => {
        responseType = 'html';
      });

      beforeEach(() => {
        res.data = {};
      });

      describe('success', () => {
        beforeEach(async () => {
          mockGetFullAddresses.mockResolvedValueOnce(mockAddresses);

          await getLandingPage(req, res, next);
        });

        it('should log correctly', () => {
          expect(logOutcome).toHaveBeenCalledWith('landing', 'successful', req);
        });

        it('should set the correct response data', () => {
          expect(res.data).toMatchSnapshot();
        });

        it('should call next correctly', () => {
          expect(next).toHaveBeenCalledWith();
        });
      });
    });

    describe('[Content-Type: json]', () => {
      beforeAll(() => {
        responseType = 'json';
      });

      beforeEach(() => {
        res.data = {};
      });

      describe('success', () => {
        beforeEach(async () => {
          mockGetFullAddresses.mockResolvedValueOnce(mockAddresses);

          await getLandingPage(req, res, next);
        });

        it('should set the correct response data', () => {
          expect(res.send.mock.calls).toMatchSnapshot();
        });
      });
    });
  });
});
