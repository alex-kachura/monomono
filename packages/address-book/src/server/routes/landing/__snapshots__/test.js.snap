// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Route: /] [GET] [Content-Type: html] success should set the correct response data 1`] = `
Object {
  "payload": Object {
    "addresses": Object {
      "other-addresses": Array [],
      "primary-addresses": Object {
        "clubcard": Object {
          "addressLines": Array [
            Object {
              "label": "Lever building",
              "lineNumber": 1,
            },
          ],
          "label": "HOME",
          "postTown": "London",
          "postcode": "EC1R 5AR",
          "tags": Array [
            "primaryLoyalty",
          ],
          "telephoneNumbers": Array [
            Object {
              "label": "Day",
              "value": "07429049112",
            },
          ],
        },
        "grocery": Object {
          "addressLines": Array [
            Object {
              "label": "Lever building",
              "lineNumber": 1,
            },
          ],
          "label": "HOME",
          "postTown": "London",
          "postcode": "EC1R 5AR",
          "tags": Array [
            "primaryDelivery",
          ],
          "telephoneNumbers": Array [
            Object {
              "label": "Day",
              "value": "07429049112",
            },
          ],
        },
      },
    },
    "banner": Object {
      "bannerType": "",
      "description": "",
      "title": "",
    },
    "breadcrumb": Array [
      Object {
        "href": "/account/mock-lang/manage",
        "text": "pages.account.title",
      },
      Object {
        "current": true,
        "text": "pages.landing.title",
      },
    ],
  },
}
`;

exports[`[Route: /] [GET] [Content-Type: json] success should set the correct response data 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "addresses": Object {
          "other-addresses": Array [],
          "primary-addresses": Object {
            "clubcard": Object {
              "addressLines": Array [
                Object {
                  "label": "Lever building",
                  "lineNumber": 1,
                },
              ],
              "label": "HOME",
              "postTown": "London",
              "postcode": "EC1R 5AR",
              "tags": Array [
                "primaryLoyalty",
              ],
              "telephoneNumbers": Array [
                Object {
                  "label": "Day",
                  "value": "07429049112",
                },
              ],
            },
            "grocery": Object {
              "addressLines": Array [
                Object {
                  "label": "Lever building",
                  "lineNumber": 1,
                },
              ],
              "label": "HOME",
              "postTown": "London",
              "postcode": "EC1R 5AR",
              "tags": Array [
                "primaryDelivery",
              ],
              "telephoneNumbers": Array [
                Object {
                  "label": "Day",
                  "value": "07429049112",
                },
              ],
            },
          },
        },
        "banner": Object {
          "bannerType": "",
          "description": "",
          "title": "",
        },
        "breadcrumb": Array [
          Object {
            "href": "/account/mock-lang/manage",
            "text": "pages.account.title",
          },
          Object {
            "current": true,
            "text": "pages.landing.title",
          },
        ],
      },
    },
  ],
]
`;
